extensions/flight-sql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -650,9 +647,8 @@ private Table executeSqlQuery(SessionState session, String sql) {
         // See SQLTODO(catalog-reader-implementation)
         final ExecutionContext executionContext = ExecutionContext.getContext();
         final QueryScope queryScope = executionContext.getQueryScope();
-        // noinspection unchecked,rawtypes
         final Map<String, Table> queryScopeTables =
-                (Map<String, Table>) (Map) queryScope.toMap(queryScope::unwrapObject, (n, t) -> t instanceof Table);
+                queryScope.toMap(o -> queryScopeAuthorizedTableMapper(queryScope, o), (n, t) -> t != null);
```

Comment:
Transform is arbitrarily expensive. It would be much better if we could only resolve the tables we need.

Reply:
Liveness-wise, we should probably:
1. retain everything we put into `queryScopeTables` (if refreshing)
2. drop anything not visited by the parents visitor after we can know that set
3. drop the remainder once we've created the result (since the result should enforce liveness to its dependencies)

Reply:
UpdateGraph selection should be based on the refreshing parents, not the in-place exec context.
So you want to use:
`io.deephaven.engine.updategraph.NotificationQueue.Dependency#getUpdateGraph(io.deephaven.engine.updategraph.NotificationQueue.Dependency...)`
or
```
io.deephaven.engine.updategraph.NotificationQueue.Dependency#getUpdateGraph(io.deephaven.engine.updategraph.NotificationQueue.Dependency, io.deephaven.engine.updategraph.NotificationQueue.Dependency...)
```
to get the UG, then *open* an exec context `withUpdateGraph`, and use that UG for the lock, etc.